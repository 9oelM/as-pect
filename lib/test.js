"use strict";
/**
 * This file is used by the package.json "asp" script to simulate using the command line interface.
 */
require("../lib/cli/index.js").asp(process.argv.slice(2));
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidGVzdC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIi4uL3NyYy90ZXN0LnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7QUFBQTs7R0FFRztBQUNILE9BQU8sQ0FBQyxxQkFBcUIsQ0FBQyxDQUFDLEdBQUcsQ0FBQyxPQUFPLENBQUMsSUFBSSxDQUFDLEtBQUssQ0FBQyxDQUFDLENBQUMsQ0FBQyxDQUFDIn0=