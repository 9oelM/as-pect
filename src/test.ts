import { asp } from "./cli";

asp(process.argv.slice(2));
