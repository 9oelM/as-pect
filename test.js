const glob = require("glob");
const path = require("path");
const pattern = path.join(process.cwd(), "assembly/tests/**/*.ts");
console.log(pattern);

console.log(glob.sync(pattern));