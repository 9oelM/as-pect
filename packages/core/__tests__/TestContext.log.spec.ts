import { TestContext } from "../src/test/TestContext";
import { createLogModule } from "./setup/createLogModule";

let ctx: TestContext;

let start = new Promise<void>((resolve, reject) => {
  createLogModule({}, (err, result) => {
    if (err) {
      console.log(err);
      reject(err);
    } else {
      ctx = result!;
      resolve();
    }
  });
});

beforeEach(() => start);

describe("log output", () => {
  test("Overall Statistics", () => {});
  for (const group of ctx.testGroups) {
    test(`Group: ${group.name}`, () => {
      for (const log of group.logs) {
        expect(log.bytes).toMatchSnapshot(`bytes`);
        expect(log.message).toMatchSnapshot(`message`);
        expect(log.offset).toMatchSnapshot(`offset`);
        expect(log.target).toBe(group);
        expect(log.value).toMatchSnapshot(`value`);
        expect(log.values).toMatchSnapshot(`values`);
      }
      for (const todo of group.todos) {
        expect(todo).toMatchSnapshot(`todo`);
      }
      expect(group.pass).toBeTruthy();
    });

    describe(`Group: ${group.name}`, () => {
      for (const groupTest of group.tests) {
        test(`Test: ${groupTest.name}`, () => {
          for (const log of groupTest.logs) {
            expect(log.bytes).toMatchSnapshot(`bytes`);
            expect(log.message).toMatchSnapshot(`message`);
            expect(log.offset).toMatchSnapshot(`offset`);
            expect(log.target).toBe(groupTest);
            expect(log.value).toMatchSnapshot(`snapshot`);
          }
          expect(groupTest.pass).toBeTruthy();
        });
      }
    });
  }
});
