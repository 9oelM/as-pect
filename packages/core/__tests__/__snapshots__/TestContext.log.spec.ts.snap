// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`log output Group: logs Test: logs: bytes 1`] = `Array []`;

exports[`log output Group: logs Test: logs: bytes 2`] = `Array []`;

exports[`log output Group: logs Test: logs: bytes 3`] = `Array []`;

exports[`log output Group: logs Test: logs: bytes 4`] = `
Array [
  0,
  0,
  0,
  0,
  0,
  0,
  240,
  63,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  64,
  0,
  0,
  0,
  0,
  0,
  0,
  8,
  64,
]
`;

exports[`log output Group: logs Test: logs: bytes 5`] = `Array []`;

exports[`log output Group: logs Test: logs: bytes 6`] = `Array []`;

exports[`log output Group: logs Test: logs: bytes 7`] = `Array []`;

exports[`log output Group: logs Test: logs: bytes 8`] = `Array []`;

exports[`log output Group: logs Test: logs: bytes 9`] = `Array []`;

exports[`log output Group: logs Test: logs: bytes 10`] = `Array []`;

exports[`log output Group: logs Test: logs: message 1`] = `"Hello world!"`;

exports[`log output Group: logs Test: logs: message 2`] = `"Value 42"`;

exports[`log output Group: logs Test: logs: message 3`] = `"null"`;

exports[`log output Group: logs Test: logs: message 4`] = `"Reference Type"`;

exports[`log output Group: logs Test: logs: message 5`] = `"[1,2,3]"`;

exports[`log output Group: logs Test: logs: message 6`] = `"Value 9999999999999"`;

exports[`log output Group: logs Test: logs: message 7`] = `"Value 1234567"`;

exports[`log output Group: logs Test: logs: message 8`] = `"Value 999999999999"`;

exports[`log output Group: logs Test: logs: message 9`] = `"Value -1"`;

exports[`log output Group: logs Test: logs: message 10`] = `"Value -1"`;

exports[`log output Group: logs Test: logs: offset 1`] = `0`;

exports[`log output Group: logs Test: logs: offset 2`] = `0`;

exports[`log output Group: logs Test: logs: offset 3`] = `0`;

exports[`log output Group: logs Test: logs: offset 4`] = `24`;

exports[`log output Group: logs Test: logs: offset 5`] = `0`;

exports[`log output Group: logs Test: logs: offset 6`] = `0`;

exports[`log output Group: logs Test: logs: offset 7`] = `0`;

exports[`log output Group: logs Test: logs: offset 8`] = `0`;

exports[`log output Group: logs Test: logs: offset 9`] = `0`;

exports[`log output Group: logs Test: logs: offset 10`] = `0`;

exports[`log output Group: logs Test: logs: snapshot 1`] = `296`;

exports[`log output Group: logs Test: logs: snapshot 2`] = `42`;

exports[`log output Group: logs Test: logs: snapshot 3`] = `null`;

exports[`log output Group: logs Test: logs: snapshot 4`] = `2144`;

exports[`log output Group: logs Test: logs: snapshot 5`] = `null`;

exports[`log output Group: logs Test: logs: snapshot 6`] = `null`;

exports[`log output Group: logs Test: logs: snapshot 7`] = `1234567`;

exports[`log output Group: logs Test: logs: snapshot 8`] = `null`;

exports[`log output Group: logs Test: logs: snapshot 9`] = `-1`;

exports[`log output Group: logs Test: logs: snapshot 10`] = `-1`;

exports[`log output Group: logs: bytes 1`] = `Array []`;

exports[`log output Group: logs: bytes 2`] = `Array []`;

exports[`log output Group: logs: bytes 3`] = `Array []`;

exports[`log output Group: logs: bytes 4`] = `
Array [
  0,
  0,
  0,
  0,
  0,
  0,
  240,
  63,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  64,
  0,
  0,
  0,
  0,
  0,
  0,
  8,
  64,
]
`;

exports[`log output Group: logs: bytes 5`] = `Array []`;

exports[`log output Group: logs: bytes 6`] = `Array []`;

exports[`log output Group: logs: bytes 7`] = `Array []`;

exports[`log output Group: logs: bytes 8`] = `Array []`;

exports[`log output Group: logs: bytes 9`] = `Array []`;

exports[`log output Group: logs: bytes 10`] = `Array []`;

exports[`log output Group: logs: message 1`] = `"Hello world!"`;

exports[`log output Group: logs: message 2`] = `"Value 42"`;

exports[`log output Group: logs: message 3`] = `"null"`;

exports[`log output Group: logs: message 4`] = `"Reference Type"`;

exports[`log output Group: logs: message 5`] = `"[1,2,3]"`;

exports[`log output Group: logs: message 6`] = `"Value 9999999999999"`;

exports[`log output Group: logs: message 7`] = `"Value 1234567"`;

exports[`log output Group: logs: message 8`] = `"Value 999999999999"`;

exports[`log output Group: logs: message 9`] = `"Value -1"`;

exports[`log output Group: logs: message 10`] = `"Value -1"`;

exports[`log output Group: logs: offset 1`] = `0`;

exports[`log output Group: logs: offset 2`] = `0`;

exports[`log output Group: logs: offset 3`] = `0`;

exports[`log output Group: logs: offset 4`] = `24`;

exports[`log output Group: logs: offset 5`] = `0`;

exports[`log output Group: logs: offset 6`] = `0`;

exports[`log output Group: logs: offset 7`] = `0`;

exports[`log output Group: logs: offset 8`] = `0`;

exports[`log output Group: logs: offset 9`] = `0`;

exports[`log output Group: logs: offset 10`] = `0`;

exports[`log output Group: logs: snapshot 1`] = `296`;

exports[`log output Group: logs: snapshot 2`] = `42`;

exports[`log output Group: logs: snapshot 3`] = `null`;

exports[`log output Group: logs: snapshot 4`] = `2144`;

exports[`log output Group: logs: snapshot 5`] = `null`;

exports[`log output Group: logs: snapshot 6`] = `null`;

exports[`log output Group: logs: snapshot 7`] = `1234567`;

exports[`log output Group: logs: snapshot 8`] = `null`;

exports[`log output Group: logs: snapshot 9`] = `-1`;

exports[`log output Group: logs: snapshot 10`] = `-1`;

exports[`log output Group: logs: todo 1`] = `"one"`;

exports[`log output Group: logs: todo 2`] = `"two"`;

exports[`log output Group: logs: todo 3`] = `"three"`;
